src/index.js

```diff
@@ -60,26 +57,18 @@ class DelegatedContentRouting {
    * @param {CID} key
    * @param {object} options
    * @param {number} options.maxTimeout How long the query can take. Defaults to 30 seconds
-   * @param {function(Error, Array<PeerInfo>)} callback
-   * @returns {void}
+   * @returns {AsyncIterable<PeerInfo>}
    */
-  findProviders (key, options, callback) {
-    if (typeof options === 'function') {
-      callback = options
-      options = {}
-    } else if (typeof options === 'number') { // This will be deprecated in a next release
-      options = {
-        maxTimeout: options
-      }
-    } else {
-      options = options || {}
-    }
-
+  async * findProviders (key, options = {}) {
```

Comment:
👍 it will be nice to have an iterable api all ready to go once the dht supports it.

Reply:
Death to lists.  Iterate all of the things.

Seriously, it's going to alleviate so many memory problems..